### DISCUSSION TITLE
Exclude children from the gesture

### DISCUSSION THREAD

**Discussion by odusseys** [pmndrs/use-gesture#428] [Q&A] [answered]
Hi,
Is there a way to exclude specific children of an element from triggering the gesture / getting events as they should ? For example, is there a way to correctly select text within a draggable elements ?

**Comment by dbismut** [comment, marked as answer]
One way of doing this: https://codesandbox.io/s/intelligent-sanderson-6ox7y

I understand the request though, if there's a more elaborate use case, we could add an option like accepting an array of nodes that would be tested against when the gesture starts and that would prevent the gesture.

**Reply by odusseys** [reply]
that's pretty much what I ended up doing - not magnificent but it works. One way you could implement this is creating a wrapper component for excluded children, which adds an attribute to the underlying node (ex use-gesture-excluded="true") ; and the lib could check for this attribute to prevent the gesture if the attribute is present on one of the parents of the target
the most react-way of doing it would be using Context but it means wrapping the entire gesture-enabled component with a provider (like in React-dnd) - that being said it would enable a lot of additional potential functionality, such as children subscribing to the gesture and being able to cancel it mid gesture, etc.
thanks for your time and your answer

**Reply by dbismut** [reply]
I see what you mean. Do you see this option accepting an array of dom notes / react refs? Or even possibly a function?
